### PR TITLE
ci: Modify docker build CI

### PR DESCRIPTION
**Pull request by rgantonio**
This PR modifies the CI of the docker build such that it builds the SNAX container rather than the original Snitch Cluster container.

The DockerfileChisel contains the updated Snitch Cluster container but also added to it is the Chisel environment as well as the new LLVM 17.

### CODE REVIEW

**Comment by rgantonio** [comment]
@JosseVanDelm the build is working LOL but you need to click it manually in the actions. I'm testing if the container is indeed correct :) by the looks of it, it does 😄 weeee~

**Comment by rgantonio** [comment]
@JosseVanDelm @xiaoling-yi @jorendumoulin approve plez HAHAHAHa

**Review by xiaoling-yi — APPROVED** [review]

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -20,13 +20,13 @@ jobs:
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          password: ${{ secrets.GH_PAT }}
```
Comment (JosseVanDelm) [inline .github/workflows/build-docker.yml:23]: Why are you using GH_PAT here? This assumes that the user has a PAT setup?
What's wrong with using GITHUB_TOKEN?

**Review by JosseVanDelm — CHANGES_REQUESTED** [review]
Looks good, apart from the password field?

File: .github/workflows/build-docker.yml (hunk 1 above)
Comment (rgantonio) [reply .github/workflows/build-docker.yml:23]: Hmm it was complaining before that I should not use GITHUB_TOKEN 😢 wait let me try to modify and see but last time it didn't want me to use the word GITHUB_TOKEN for some reason. So I just used GH_PAT. I'll rename it wait.

Comment (rgantonio) [reply .github/workflows/build-docker.yml:23]: Yeah, github complains:

image

Comment (rgantonio) [reply .github/workflows/build-docker.yml]: @JosseVanDelm seems to build OKAY 😄  cause it logged in normally. Changed it to `GH_TOKEN` instead!

Comment (JosseVanDelm) [reply .github/workflows/build-docker.yml]: That is so weird it totally works here:
https://github.com/KULeuven-MICAS/snax-mlir/blob/main/.github/workflows/build-docker.yml

Anyways if it works it works I guess

**Review by JosseVanDelm — APPROVED** [review]
Very nice thank you!

**Comment by rgantonio** [comment]
Thank you @JosseVanDelm and @xiaoling-yi! Mwah mwah chup chup hahaha

**Pull request merged**